### PR TITLE
[Tests-Only][full-ci] enforce strict type checking for php api tests

### PR DESCRIPTION
**Pull request by dpakach**
## Description

Enable strict type checking in local API tests.

## Related Issue

- Related https://github.com/owncloud/QA/issues/694

## How Has This Been Tested?

:robot:

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

**Comment by ownclouders** [comment]
:boom: Acceptance test [<strong>localApiTests-apiSpaces-ocis</strong>](https://drone.owncloud.com/owncloud/ocis/7891/20/1) failed. Further test are cancelled...

File: tests/acceptance/features/bootstrap/SpacesContext.php [hunk 1]
```diff
@@ -163,7 +163,7 @@ public function getSpaceIdByNameFromResponse(string $name): string {
 	 * @throws Exception
 	 */
 	public function getSpaceByNameFromResponse(string $name): array {
-		$response = json_decode($this->featureContext->getResponse()->getBody(), true, 512, JSON_THROW_ON_ERROR);
+		$response = json_decode((string)$this->featureContext->getResponse()->getBody(), true, 512, JSON_THROW_ON_ERROR);
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/SpacesContext.php:166]:
```suggestion
		$response = json_decode($this->featureContext->getResponse()->getBody()->getContents(), true, 512, JSON_THROW_ON_ERROR);
```

Can we do this to always get a string?

File: tests/acceptance/features/bootstrap/SpacesContext.php [hunk 2]
```diff
@@ -434,7 +434,7 @@ public function theAdministratorGivesUserTheRole(string $user, string $role): vo
 		$fullUrl = $baseUrl . "api/v0/settings/roles-list";
 		$this->featureContext->setResponse(HttpRequestHelper::post($fullUrl, "", $admin, $password, $headers, "{}"));
 		if ($this->featureContext->getResponse()) {
-			$rawBody =  $this->featureContext->getResponse()->getBody()->getContents();
+			$rawBody =  (string)$this->featureContext->getResponse()->getBody()->getContents();
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/SpacesContext.php:437]: Doesn't this already return a string?

**Review by phil-davis — COMMENTED** [review]
Just 2 questions.

File: tests/acceptance/features/bootstrap/SpacesContext.php (hunk 1 above)
Comment (dpakach) [reply tests/acceptance/features/bootstrap/SpacesContext.php:166]: I tried doing this but the tests failed.
https://drone.owncloud.com/owncloud/ocis/8104/20/6

**Review by phil-davis — APPROVED** [review]

**Pull request merged**